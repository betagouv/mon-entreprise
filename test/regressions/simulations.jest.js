// The goal of these tests is to avoid deploying unwanted changes in the calculations. We run a number
// of simulations and persist their results in a snapshot (ie, a file commited in git). Our test runner,
// Jest, then compare the existing snapshot with the current Engine calculation and reports any difference.
//
// We only persist targets values in the file system, in order to be resilient to rule renaming (if a rule is
// renamed the test configuration may be adapted but the persisted snapshot will remain unchanged).

/* eslint-disable no-undef */
import artisteAuteurConfig from '../../source/components/simulationConfigs/artiste-auteur.yaml'
import autoentrepreneurConfig from '../../source/components/simulationConfigs/auto-entrepreneur.yaml'
import independantConfig from '../../source/components/simulationConfigs/indépendant.yaml'
import remunerationDirigeantConfig from '../../source/components/simulationConfigs/rémunération-dirigeant.yaml'
import employeeConfig from '../../source/components/simulationConfigs/salarié.yaml'
import Engine from '../../source/engine'
import artisteAuteurSituations from './simulations-artiste-auteur.yaml'
import autoEntrepreneurSituations from './simulations-auto-entrepreneur.yaml'
import independentSituations from './simulations-indépendant.yaml'
import remunerationDirigeantSituations from './simulations-rémunération-dirigeant.yaml'
import employeeSituations from './simulations-salarié.yaml'

const roundResult = arr => arr.map(x => Math.round(x))
const engine = new Engine()
const runSimulations = (
	situations,
	targets,
	baseSituation = {},
	defaultUnits,
	namePrefix = ''
) =>
	Object.entries(situations).map(([name, situations]) =>
		situations.forEach(situation => {
			engine.setSituation({ ...baseSituation, ...situation })
			engine.setDefaultUnits(defaultUnits)
			const res = engine.evaluate(targets).map(node => node.nodeValue)
			// Stringify is not required, but allows the result to be displayed in a single
			// line in the snapshot, which considerably reduce the number of lines of this snapshot
			// and improve its readability.
			expect(JSON.stringify(roundResult(res))).toMatchSnapshot(
				namePrefix + ' ' + name
			)
		})
	)

it('calculate simulations-salarié', () => {
	runSimulations(
		employeeSituations,
		employeeConfig.objectifs,
		employeeConfig.situation,
		['€/mois']
	)
})

it('calculate simulations-indépendant', () => {
	const targets = independantConfig.objectifs.reduce(
		(acc, cur) => [...acc, ...cur.objectifs],
		[]
	)
	runSimulations(independentSituations, targets, independantConfig.situation, [
		'€/an'
	])
})

it('calculate simulations-auto-entrepreneur', () => {
	runSimulations(
		autoEntrepreneurSituations,
		autoentrepreneurConfig.objectifs,
		autoentrepreneurConfig.situation,
		['€/an']
	)
})

it('calculate simulations-rémunération-dirigeant', () => {
	const baseSituation = remunerationDirigeantConfig.situation
	remunerationDirigeantConfig.branches.forEach(({ nom, situation }) => {
		runSimulations(
			remunerationDirigeantSituations,
			remunerationDirigeantConfig.objectifs,
			{ ...baseSituation, ...situation },
			['€/an'],
			`${nom} - `
		)
	})
})

it('calculate simulations-artiste-auteur', () => {
	runSimulations(
		artisteAuteurSituations,
		artisteAuteurConfig.objectifs,
		artisteAuteurConfig.situation,
		['€/an']
	)
})
