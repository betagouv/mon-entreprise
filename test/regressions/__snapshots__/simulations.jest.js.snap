// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calculate simulations-artiste-auteur:  bnc 1`] = `"[1230]"`;

exports[`calculate simulations-artiste-auteur:  bnc 2`] = `"[1230]"`;

exports[`calculate simulations-artiste-auteur:  bnc 3`] = `"[1230]"`;

exports[`calculate simulations-artiste-auteur:  salarié 1`] = `"[160]"`;

exports[`calculate simulations-artiste-auteur:  salarié 2`] = `"[1603]"`;

exports[`calculate simulations-artiste-auteur:  salarié 3`] = `"[12410]"`;

exports[`calculate simulations-auto-entrepreneur:  ACRE 1`] = `"[21394,1394,20000,0,20000]"`;

exports[`calculate simulations-auto-entrepreneur:  ACRE 2`] = `"[31029,1029,30000,0,30000]"`;

exports[`calculate simulations-auto-entrepreneur:  ACRE 3`] = `"[44304,4304,40000,0,40000]"`;

exports[`calculate simulations-auto-entrepreneur:  aides 1`] = `"[5348,348,5000,0,5000]"`;

exports[`calculate simulations-auto-entrepreneur:  aides 2`] = `"[53485,3485,50000,93,49907]"`;

exports[`calculate simulations-auto-entrepreneur:  impôt sur le revenu 1`] = `"[32092,7092,25000,706,24294]"`;

exports[`calculate simulations-auto-entrepreneur:  échelle de revenus 1`] = `"[574,74,500,0,500]"`;

exports[`calculate simulations-auto-entrepreneur:  échelle de revenus 2`] = `"[1148,148,1000,0,1000]"`;

exports[`calculate simulations-auto-entrepreneur:  échelle de revenus 3`] = `"[2297,297,2000,0,2000]"`;

exports[`calculate simulations-auto-entrepreneur:  échelle de revenus 4`] = `"[5742,742,5000,0,5000]"`;

exports[`calculate simulations-auto-entrepreneur:  échelle de revenus 5`] = `"[11483,1483,10000,0,10000]"`;

exports[`calculate simulations-auto-entrepreneur:  échelle de revenus 6`] = `"[22966,2966,20000,0,20000]"`;

exports[`calculate simulations-auto-entrepreneur:  échelle de revenus 7`] = `"[57415,7415,50000,275,49725]"`;

exports[`calculate simulations-auto-entrepreneur:  échelle de revenus 8`] = `"[80381,10381,70000,1340,68660]"`;

exports[`calculate simulations-auto-entrepreneur:  échelle de revenus 9`] = `"[114830,14830,100000,4008,95992]"`;

exports[`calculate simulations-auto-entrepreneur:  échelle de revenus 10`] = `"[1148303,148303,1000000,131979,868021]"`;

exports[`calculate simulations-indépendant:  acre 1`] = `"[73024,23024,50000,51980,8052,41948,null,73024]"`;

exports[`calculate simulations-indépendant:  activité 1`] = `"[28923,8923,20000,20783,604,19396,null,28923]"`;

exports[`calculate simulations-indépendant:  activité 2`] = `"[29101,9101,20000,20787,604,19396,null,29101]"`;

exports[`calculate simulations-indépendant:  impôt sur le revenu 1`] = `"[29085,9085,20000,20787,603,19397,null,29085]"`;

exports[`calculate simulations-indépendant:  impôt sur le revenu 2`] = `"[73024,23024,50000,51980,8213,41787,null,73024]"`;

exports[`calculate simulations-indépendant:  impôt sur le revenu 3`] = `"[29085,9085,20000,20787,2079,17921,null,29085]"`;

exports[`calculate simulations-indépendant:  inversions 1`] = `"[2000,1384,616,667,0,616,null,2000]"`;

exports[`calculate simulations-indépendant:  inversions 2`] = `"[50000,16003,33997,35352,3563,30434,null,50000]"`;

exports[`calculate simulations-indépendant:  inversions 3`] = `"[14596,4596,10000,10394,0,10000,null,14596]"`;

exports[`calculate simulations-indépendant:  inversions 4`] = `"[88547,27360,61187,63588,11187,50000,null,88547]"`;

exports[`calculate simulations-indépendant:  inversions 5`] = `"[14596,4596,10000,10394,0,10000,null,15596]"`;

exports[`calculate simulations-indépendant:  inversions 6`] = `"[19000,5928,13072,13585,0,13072,1000,20000]"`;

exports[`calculate simulations-indépendant:  inversions 7`] = `"[18000,5626,12374,12860,0,12374,2000,20000]"`;

exports[`calculate simulations-indépendant:  échelle de revenus 1`] = `"[1859,1359,500,548,0,500,null,1859]"`;

exports[`calculate simulations-indépendant:  échelle de revenus 2`] = `"[2467,1467,1000,1064,0,1000,null,2467]"`;

exports[`calculate simulations-indépendant:  échelle de revenus 3`] = `"[3075,1575,1500,1581,0,1500,null,3075]"`;

exports[`calculate simulations-indépendant:  échelle de revenus 4`] = `"[3682,1682,2000,2097,0,2000,null,3682]"`;

exports[`calculate simulations-indépendant:  échelle de revenus 5`] = `"[7427,2427,5000,5199,0,5000,null,7427]"`;

exports[`calculate simulations-indépendant:  échelle de revenus 6`] = `"[14596,4596,10000,10394,0,10000,null,14596]"`;

exports[`calculate simulations-indépendant:  échelle de revenus 7`] = `"[139594,39594,100000,103788,24245,75755,null,139594]"`;

exports[`calculate simulations-indépendant:  échelle de revenus 8`] = `"[1239955,239955,1000000,1033666,467505,532495,null,1239955]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  ACRE 1`] = `"[7257,7257,7184,4,13,16]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  ACRE 2`] = `"[14963,14963,14544,4,26,32]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  ACRE 3`] = `"[21575,22669,21905,4,40,48]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  activités 1`] = `"[10982,10982,10742,4,19,23]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  activités 2`] = `"[10982,10982,10742,4,19,23]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  activités 3`] = `"[10982,10982,10742,4,19,23]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  activités 4`] = `"[10982,10982,10742,4,19,23]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  activités 5`] = `"[10982,10982,10742,4,19,23]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  avec charges 1`] = `"[5291,5291,5306,4,10,12]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  avec charges 2`] = `"[10982,10982,10742,4,19,23]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  échelle de rémunération 1`] = `"[-2313,-2313,2488,0,0,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  échelle de rémunération 2`] = `"[169,169,139,0,1,1]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  échelle de rémunération 3`] = `"[738,738,323,0,2,2]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  échelle de rémunération 4`] = `"[2446,2446,2588,2,5,6]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  échelle de rémunération 5`] = `"[5291,5291,5306,4,10,12]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  échelle de rémunération 6`] = `"[10982,10982,10742,4,19,23]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  échelle de rémunération 7`] = `"[25971,28055,27050,4,46,59]"`;

exports[`calculate simulations-rémunération-dirigeant: Assimilé salarié -  échelle de rémunération 8`] = `"[46812,57017,52684,4,46,119]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  ACRE 1`] = `"[9349,9349,2046,2,0,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  ACRE 2`] = `"[18697,18697,4093,3,8,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  ACRE 3`] = `"[28046,28046,6139,4,12,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  activités 1`] = `"[15580,15580,6600,4,18,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  activités 2`] = `"[15560,15560,0,4,0,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  activités 3`] = `"[17336,17336,4093,3,8,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  activités 4`] = `"[17417,17417,4093,3,8,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  activités 5`] = `"[17417,17417,4093,3,8,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  avec charges 1`] = `"[8450,8450,2456,3,0,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  avec charges 2`] = `"[15480,15480,7163,4,14,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  échelle de rémunération 1`] = `"[87,87,6,0,0,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  échelle de rémunération 2`] = `"[871,871,60,0,0,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  échelle de rémunération 3`] = `"[1742,1742,119,0,0,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  échelle de rémunération 4`] = `"[4354,4354,1023,1,0,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  échelle de rémunération 5`] = `"[8709,8709,2046,2,0,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  échelle de rémunération 6`] = `"[17417,17417,4093,3,8,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  échelle de rémunération 7`] = `"[43543,43543,10232,4,20,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Auto-entrepreneur -  échelle de rémunération 8`] = `"[84367,87085,20465,4,40,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  ACRE 1`] = `"[8215,8215,6018,4,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  ACRE 2`] = `"[16442,16527,12103,4,24,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  ACRE 3`] = `"[20731,21570,15799,4,31,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  activités 1`] = `"[13886,13886,10166,4,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  activités 2`] = `"[14645,14645,0,4,0,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  activités 3`] = `"[13758,13758,10075,4,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  activités 4`] = `"[13769,13769,10084,4,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  activités 5`] = `"[13769,13769,10084,4,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  avec charges 1`] = `"[6795,6795,4977,4,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  avec charges 2`] = `"[13769,13769,10084,4,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  échelle de rémunération 1`] = `"[-1044,-1044,0,3,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  échelle de rémunération 2`] = `"[-225,-225,0,3,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  échelle de rémunération 3`] = `"[616,616,470,3,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  échelle de rémunération 4`] = `"[3084,3084,2267,3,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  échelle de rémunération 5`] = `"[6795,6795,4977,4,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  échelle de rémunération 6`] = `"[13769,13769,10084,4,21,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  échelle de rémunération 7`] = `"[30434,33997,24912,4,48,0]"`;

exports[`calculate simulations-rémunération-dirigeant: Indépendant -  échelle de rémunération 8`] = `"[56273,69895,36431,4,56,0]"`;

exports[`calculate simulations-salarié:  JEI 1`] = `"[3440,0,0,3000,2353,2187]"`;

exports[`calculate simulations-salarié:  JEI 2`] = `"[26635,0,0,20000,15969,10681]"`;

exports[`calculate simulations-salarié:  JEI 3`] = `"[4517,0,0,4000,3141,2741]"`;

exports[`calculate simulations-salarié:  activité partielle 1`] = `"[27,1218,0,1560,1197,1197]"`;

exports[`calculate simulations-salarié:  activité partielle 2`] = `"[27,2800,0,4000,2594,2354]"`;

exports[`calculate simulations-salarié:  activité partielle 3`] = `"[778,4849,0,8000,5209,4232]"`;

exports[`calculate simulations-salarié:  activité partielle 4`] = `"[850,2240,0,4000,2704,2435]"`;

exports[`calculate simulations-salarié:  activité partielle 5`] = `"[2479,1400,0,4000,2869,2556]"`;

exports[`calculate simulations-salarié:  activité partielle 6`] = `"[27,2100,3750,3000,1940,1844]"`;

exports[`calculate simulations-salarié:  activité partielle 7`] = `"[27,2800,0,4000,2594,2491]"`;

exports[`calculate simulations-salarié:  activité partielle 8`] = `"[227,1400,0,2000,1540,1515]"`;

exports[`calculate simulations-salarié:  activité partielle 9`] = `"[1156,700,0,2000,1551,1519]"`;

exports[`calculate simulations-salarié:  activité partielle 10`] = `"[327,4200,0,6000,4182,3494]"`;

exports[`calculate simulations-salarié:  aides 1`] = `"[2302,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  aides 2`] = `"[12823,0,0,10000,8911,7667]"`;

exports[`calculate simulations-salarié:  aides 3`] = `"[2062,417,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  aides 4`] = `"[2292,208,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  apprentissage 1`] = `"[1552,0,0,1500,1448,1448]"`;

exports[`calculate simulations-salarié:  apprentissage 2`] = `"[1385,167,0,1500,1448,1448]"`;

exports[`calculate simulations-salarié:  assimilé salarié 1`] = `"[7015,0,0,5000,3943,3318]"`;

exports[`calculate simulations-salarié:  assimilé salarié 2`] = `"[1583,0,0,1500,1163,1163]"`;

exports[`calculate simulations-salarié:  assimilé salarié 3`] = `"[3685,0,0,3000,2348,2172]"`;

exports[`calculate simulations-salarié:  atmp 1`] = `"[2534,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  avantages 1`] = `"[2667,0,0,2000,1540,1492]"`;

exports[`calculate simulations-salarié:  avantages 2`] = `"[2677,0,0,2000,1539,1490]"`;

exports[`calculate simulations-salarié:  avantages 3`] = `"[2587,0,0,2000,1549,1506]"`;

exports[`calculate simulations-salarié:  cadre 1`] = `"[4122,0,0,3000,2348,2171]"`;

exports[`calculate simulations-salarié:  cdd 1`] = `"[2509,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  cdd 2`] = `"[2591,0,0,2000,1599,1557]"`;

exports[`calculate simulations-salarié:  cdd 3`] = `"[3394,0,0,2400,1967,1883]"`;

exports[`calculate simulations-salarié:  effectif 1`] = `"[2479,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  effectif 2`] = `"[2525,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  effectif 3`] = `"[2539,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  effectif 4`] = `"[2539,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  frais pro - DFS 1`] = `"[2242,0,0,2000,1630,1630]"`;

exports[`calculate simulations-salarié:  frais pro - DFS 2`] = `"[2335,0,0,2000,1584,1544]"`;

exports[`calculate simulations-salarié:  frais pro - DFS 3`] = `"[2265,0,0,2000,1606,1563]"`;

exports[`calculate simulations-salarié:  frais pro - DFS 4`] = `"[2243,0,0,2000,1613,1569]"`;

exports[`calculate simulations-salarié:  frais pro - DFS 5`] = `"[2437,0,0,2000,1590,1590]"`;

exports[`calculate simulations-salarié:  frais pro - DFS 6`] = `"[1767,0,0,1700,1364,1364]"`;

exports[`calculate simulations-salarié:  frais pro - DFS 7`] = `"[3287,0,0,2600,2125,2097]"`;

exports[`calculate simulations-salarié:  frais pro - IKV 1`] = `"[4367,0,0,3200,2530,2320]"`;

exports[`calculate simulations-salarié:  frais pro - IKV 2`] = `"[4346,0,0,3200,2511,2302]"`;

exports[`calculate simulations-salarié:  frais pro - IKV 3`] = `"[2774,0,0,2157,1685,1630]"`;

exports[`calculate simulations-salarié:  frais pro - titres restaurant 1`] = `"[2519,0,0,2000,1521,1484]"`;

exports[`calculate simulations-salarié:  frais pro - titres restaurant 2`] = `"[4307,0,0,3000,2134,1949]"`;

exports[`calculate simulations-salarié:  frais pro - titres restaurant 3`] = `"[2562,0,0,2000,1493,1456]"`;

exports[`calculate simulations-salarié:  heures supplémentaires et complémentaires 1`] = `"[2583,0,0,2000,1636,1599]"`;

exports[`calculate simulations-salarié:  heures supplémentaires et complémentaires 2`] = `"[3105,0,0,2000,2009,1965]"`;

exports[`calculate simulations-salarié:  heures supplémentaires et complémentaires 3`] = `"[2654,0,0,2000,1636,1599]"`;

exports[`calculate simulations-salarié:  heures supplémentaires et complémentaires 4`] = `"[2565,0,0,2000,1627,1590]"`;

exports[`calculate simulations-salarié:  heures supplémentaires et complémentaires 5`] = `"[3025,0,0,2000,1970,1932]"`;

exports[`calculate simulations-salarié:  heures supplémentaires et complémentaires 6`] = `"[3336,0,2446,2000,1919,1889]"`;

exports[`calculate simulations-salarié:  heures supplémentaires et complémentaires 7`] = `"[3286,0,2286,2000,1889,1859]"`;

exports[`calculate simulations-salarié:  impôt sur le revenu 1`] = `"[4076,0,0,3000,2353,2168]"`;

exports[`calculate simulations-salarié:  impôt sur le revenu 2`] = `"[41834,0,0,30000,24227,14588]"`;

exports[`calculate simulations-salarié:  impôt sur le revenu 3`] = `"[4111,0,0,3000,2353,2172]"`;

exports[`calculate simulations-salarié:  impôt sur le revenu 4`] = `"[3896,0,0,3000,2353,2252]"`;

exports[`calculate simulations-salarié:  impôt sur le revenu 5`] = `"[41834,0,0,30000,24227,14588]"`;

exports[`calculate simulations-salarié:  impôt sur le revenu 6`] = `"[4076,0,0,3000,2626,2481]"`;

exports[`calculate simulations-salarié:  impôt sur le revenu 7`] = `"[41834,0,0,30000,26966,16204]"`;

exports[`calculate simulations-salarié:  impôt sur le revenu 8`] = `"[4076,0,0,3000,2353,2107]"`;

exports[`calculate simulations-salarié:  inversions 1`] = `"[2000,0,0,1746,1360,1353]"`;

exports[`calculate simulations-salarié:  inversions 2`] = `"[3474,0,0,2554,2000,1898]"`;

exports[`calculate simulations-salarié:  inversions 3`] = `"[3679,0,0,2706,2120,2000]"`;

exports[`calculate simulations-salarié:  lodeom 1`] = `"[1592,0,0,1521,1182,1182]"`;

exports[`calculate simulations-salarié:  lodeom 2`] = `"[2085,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  lodeom 3`] = `"[3896,0,0,3000,2353,2187]"`;

exports[`calculate simulations-salarié:  lodeom 4`] = `"[5674,0,0,4000,3146,2759]"`;

exports[`calculate simulations-salarié:  lodeom 5`] = `"[7889,0,0,5500,4349,3625]"`;

exports[`calculate simulations-salarié:  lodeom compétitivité renforcée 1`] = `"[1592,0,0,1521,1182,1182]"`;

exports[`calculate simulations-salarié:  lodeom compétitivité renforcée 2`] = `"[2085,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  lodeom compétitivité renforcée 3`] = `"[3444,0,0,3000,2353,2187]"`;

exports[`calculate simulations-salarié:  lodeom compétitivité renforcée 4`] = `"[5588,0,0,4000,3146,2759]"`;

exports[`calculate simulations-salarié:  lodeom compétitivité renforcée 5`] = `"[7889,0,0,5500,4349,3625]"`;

exports[`calculate simulations-salarié:  lodeom innovation et croissance 1`] = `"[1592,0,0,1521,1182,1182]"`;

exports[`calculate simulations-salarié:  lodeom innovation et croissance 2`] = `"[2085,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  lodeom innovation et croissance 3`] = `"[3235,0,0,3000,2353,2187]"`;

exports[`calculate simulations-salarié:  lodeom innovation et croissance 4`] = `"[4915,0,0,4000,3146,2759]"`;

exports[`calculate simulations-salarié:  lodeom innovation et croissance 5`] = `"[7889,0,0,5500,4349,3625]"`;

exports[`calculate simulations-salarié:  stage 1`] = `"[507,0,0,500,500,500]"`;

exports[`calculate simulations-salarié:  stage 2`] = `"[2490,0,0,2000,1750,1750]"`;

exports[`calculate simulations-salarié:  taux spécifiques retraite complémentaire 1`] = `"[1606,0,0,1521,1195,1195]"`;

exports[`calculate simulations-salarié:  taux spécifiques retraite complémentaire 2`] = `"[3423,0,0,2500,1979,1880]"`;

exports[`calculate simulations-salarié:  taux spécifiques retraite complémentaire 3`] = `"[1592,0,0,1521,1170,1170]"`;

exports[`calculate simulations-salarié:  taux spécifiques retraite complémentaire 4`] = `"[3382,0,0,2500,1938,1844]"`;

exports[`calculate simulations-salarié:  temps partiel 1`] = `"[2592,0,2188,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  temps partiel 2`] = `"[2533,0,2500,1857,1448,1428]"`;

exports[`calculate simulations-salarié:  temps partiel 3`] = `"[1159,0,1750,1000,770,770]"`;

exports[`calculate simulations-salarié:  treizième mois 1`] = `"[3390,0,0,2300,1950,1856]"`;

exports[`calculate simulations-salarié:  treizième mois 2`] = `"[3800,0,2965,2300,2186,2073]"`;

exports[`calculate simulations-salarié:  treizième mois 3`] = `"[3044,0,0,2300,1799,1726]"`;

exports[`calculate simulations-salarié:  échelle de salaires 1`] = `"[130,0,0,100,57,57]"`;

exports[`calculate simulations-salarié:  échelle de salaires 2`] = `"[284,0,0,250,176,176]"`;

exports[`calculate simulations-salarié:  échelle de salaires 3`] = `"[541,0,0,500,374,374]"`;

exports[`calculate simulations-salarié:  échelle de salaires 4`] = `"[799,0,0,750,572,572]"`;

exports[`calculate simulations-salarié:  échelle de salaires 5`] = `"[1056,0,0,1000,770,770]"`;

exports[`calculate simulations-salarié:  échelle de salaires 6`] = `"[1313,0,0,1250,968,968]"`;

exports[`calculate simulations-salarié:  échelle de salaires 7`] = `"[1571,0,0,1500,1165,1165]"`;

exports[`calculate simulations-salarié:  échelle de salaires 8`] = `"[2479,0,0,2000,1561,1524]"`;

exports[`calculate simulations-salarié:  échelle de salaires 9`] = `"[3401,0,0,2500,1957,1861]"`;

exports[`calculate simulations-salarié:  échelle de salaires 10`] = `"[4076,0,0,3000,2353,2187]"`;

exports[`calculate simulations-salarié:  échelle de salaires 11`] = `"[5674,0,0,4000,3146,2759]"`;

exports[`calculate simulations-salarié:  échelle de salaires 12`] = `"[7085,0,0,5000,3948,3336]"`;

exports[`calculate simulations-salarié:  échelle de salaires 13`] = `"[14319,0,0,10000,7958,6080]"`;

exports[`calculate simulations-salarié:  échelle de salaires 14`] = `"[28345,0,0,20000,15969,10681]"`;

exports[`calculate simulations-salarié:  échelle de salaires 15`] = `"[128575,0,0,100000,87157,46271]"`;

exports[`calculate simulations-salarié:  échelle de salaires 16`] = `"[1243819,0,0,1000000,896257,446123]"`;
